lib/app/features/posts/data/post_client.dart

```diff
@@ -25,7 +36,33 @@ PostClient postClient(PostClientRef _) {
 //coverage:ignore-end
 
 @riverpod
-FutureOr<List<Post>> getPosts(GetPostsRef ref) async {
+FutureOr<PostResponse> getPosts(GetPostsRef ref, PostQueryData postQueryData) async {
   final client = ref.watch(postClientProvider);
-  return client.getPosts(100);
+
+  final cancelToken = CancelToken();
+  final link = ref.keepAlive();
+
+  Timer? timer;
+
+  ref.onDispose(() {
+    cancelToken.cancel();
+    timer?.cancel();
+  });
+  ref.onCancel(() {
+    timer = Timer(PostClient.postTimeout, () {
+      // coverage:ignore-start
+      link.close();
```

Comment:
plz use ignore line

Reply:
Ignore line sometimes doesn't work for me for some reason, so that's why I didn't use it

Reply:
Nevermind, figured out how to make it work